cmd2/cmd2.py

```diff
@@ -1189,170 +1187,364 @@ def visible_prompt(self) -> str:
+        can_block = not (self._redirecting or self.in_pyscript() or self.in_script())
 
-        # Don't attempt to use a pager that can block if redirecting or running a script (either text or Python).
-        # Also only attempt to use a pager if actually running in a real fully functional terminal.
-        if functional_terminal and not self._redirecting and not self.in_pyscript() and not self.in_script():
-            final_msg = f"{msg}{end}"
-            if rich_utils.allow_style == rich_utils.AllowStyle.NEVER:
-                final_msg = ansi.strip_style(final_msg)
+        # Check if we are outputting to a pager.
+        if functional_terminal and can_block:  # pragma: no cover
+            prepared_objects = rich_utils.prepare_objects_for_rich_print(*objects)
 
-            pager = self.pager
+            # Chopping overrides soft_wrap
             if chop:
-                pager = self.pager_chop
+                soft_wrap = True
+
+            # Generate the bytes to send to the pager
+            console = Cmd2Console(self.stdout)
+            with console.capture() as capture:
+                console.print(
+                    *prepared_objects,
+                    sep=sep,
+                    end=end,
+                    style=style,
+                    soft_wrap=soft_wrap,
+                    **(rich_print_kwargs if rich_print_kwargs is not None else {}),
+                )
+            output_bytes = capture.get().encode('utf-8', 'replace')
 
-            try:
-                # Prevent KeyboardInterrupts while in the pager. The pager application will
-                # still receive the SIGINT since it is in the same process group as us.
-                with self.sigint_protection:
-                    import subprocess
-
-                    pipe_proc = subprocess.Popen(pager, shell=True, stdin=subprocess.PIPE, stdout=self.stdout)  # noqa: S602
-                    pipe_proc.communicate(final_msg.encode('utf-8', 'replace'))
-            except BrokenPipeError:
-                # This occurs if a command's output is being piped to another process and that process closes before the
-                # command is finished. If you would like your application to print a warning message, then set the
-                # broken_pipe_warning attribute to the message you want printed.`
-                if self.broken_pipe_warning:
-                    sys.stderr.write(self.broken_pipe_warning)
-        else:
-            self.poutput(msg, end=end)
+            # Prevent KeyboardInterrupts while in the pager. The pager application will
+            # still receive the SIGINT since it is in the same process group as us.
+            with self.sigint_protection:
+                import subprocess
 
-    def ppretty(self, data: Any, *, indent: int = 2, width: int = 80, depth: Optional[int] = None, end: str = '\n') -> None:
```

Comment:
Since `ppretty` is gone, we need to mention this in the breaking changes in the `CHANGELOG` and we should also add an example showing how to do pretty printing of objects like dicts using both `rich` and `pprint`

Reply:
I'll add an example at a later date once I learn the Rich pretty print API.